pipelines/curate_bridge_materialize.py

```
@@ -0,0 +1,366 @@
+        source.rename(destination)
+    except FileExistsError as exc:
+        raise BridgeCurationError(
+            f"destination already exists; refusing overwrite: {destination}"
+        ) from exc
+
+
+def _rename_posix(source: Path, destination: Path) -> None:
+    libc = ctypes.CDLL(None, use_errno=True)
+    renameat2 = getattr(libc, "renameat2", None)
+    if renameat2 is None:
+        raise BridgeCurationError("atomic no-replace publication is unavailable on this platform")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)